File: 11-container-with-most-water/container-with-most-water.cpp
class Solution {
public:
    int maxArea(vector<int>& height) {
        int n = height.size();
        int left = 0, right = n-1;
        long area = 0;
        long  maxarea = 0;

        while(left<right){
            area = min(height[left], height[right] ) * (right- left );
            
            if(height[left]<height[right]){
                left++;
            }
            else{
                right--;
            }
            maxarea = max(area, maxarea);
        }
        return maxarea;

        
    }
};

File: 11-container-with-most-water/container-with-most-water.java
class Solution {
    public int maxArea(int[] height) {
        int p1 = 0;
        int p2 = height.length - 1;
        int area = 0;
        while(p1 < p2){
            area = Math.max(area, Math.min(height[p1], height[p2])*(p2 - p1));

            if(height[p1] < height[p2]){
                p1++;
            }
            else{
                p2--;
            }
        }
        return area;
    }
}